enterprise_knowledge_retriever/tests/ekr_test.py

```diff
@@ -0,0 +1,137 @@
+It parses documents using the SambaParse service, creates a vector store, and tests the question-answering capabilities.
+
+Usage:
+    python tests/ekr_test.py
+
+Returns:
+    0 if all tests pass, or a positive integer representing the number of failed tests.
+"""
+
+import os
+import sys
+import shutil
+import time
+import unittest
+import logging
+from typing import List, Dict, Any
+
+# Setup logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+# Setup paths
+current_dir = os.getcwd()
+kit_dir = current_dir
+repo_dir = os.path.abspath(os.path.join(kit_dir, ".."))
+logger.info(f'kit_dir: {kit_dir}')
+logger.info(f'repo_dir: {repo_dir}')
+
+sys.path.append(kit_dir)
+sys.path.append(repo_dir)
+
+from src.document_retrieval import DocumentRetrieval
+from utils.parsing.sambaparse import parse_doc_universal
+
+CONFIG_PATH = os.path.join(kit_dir, 'config.yaml')
+PERSIST_DIRECTORY = os.path.join(kit_dir, "tests", "vectordata", "my-vector-db")
+TEST_DATA_PATH = os.path.join(kit_dir, "tests", "data", "test")
+
+# Let's use this as a template for further CLI tests. setup, tests, teardown and assert at the end.
+class EKRTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.time_start = time.time()
+        cls.document_retrieval = DocumentRetrieval()
+        cls.additional_metadata: Dict[str, Any] = {}
+        cls.text_chunks = cls.parse_documents()
+        cls.embeddings = cls.document_retrieval.load_embedding_model()
+        cls.vectorstore = cls.create_vectorstore()
+        cls.conversation = cls.create_conversation_chain()
+
+    @classmethod
+    def parse_documents(cls):
+        _, _, text_chunks = parse_doc_universal(doc=TEST_DATA_PATH, additional_metadata=cls.additional_metadata)
+        logger.info(f'Number of chunks: {len(text_chunks)}')
+        return text_chunks
+
+    @classmethod
+    def create_vectorstore(cls):
+        if os.path.exists(PERSIST_DIRECTORY):
+            shutil.rmtree(PERSIST_DIRECTORY)
+            logger.info("The directory Chroma has been deleted.")
+        return cls.document_retrieval.create_vector_store(cls.text_chunks, cls.embeddings, output_db=PERSIST_DIRECTORY)
+
+    @classmethod
+    def create_conversation_chain(cls):
+        cls.document_retrieval.init_retriever(cls.vectorstore)
+        return cls.document_retrieval.get_qa_retrieval_chain()
+
+    def test_document_parsing(self):
+        self.assertTrue(len(self.text_chunks) > 0, f"Parsed {len(self.text_chunks)} chunks")
+
+    def test_vector_store_creation(self):
+        self.assertIsNotNone(self.vectorstore, "Vector store created successfully")
+
+    def test_conversation_chain_creation(self):
+        self.assertIsNotNone(self.conversation, "Conversation chain created successfully")
+
+    def test_question_answering(self):
+        user_question = "What is a composition of experts?"
```

Comment:
Since we changed the PDF (I was using earlier the CoE-SN40L arxvix paper), we can change the question to:
"What is the new record set by Samba-1 Turbo?"